Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -374,10 +381,24 @@ - (void) _sendWebSocketRequest {
     // Construct the HTTP request:
     for (Dict::iterator header(_options[kC4ReplicatorOptionExtraHeaders].asDict()); header; ++header)
         _logic[slice2string(header.keyString())] = slice2string(header.value().asString());
-    slice cookies = _options[kC4ReplicatorOptionCookies].asString();
-    if (cookies)
-        [_logic addValue: cookies.asNSString() forHTTPHeaderField: @"Cookie"];
-
+    
+    slice sessionCookie = _options[kC4ReplicatorOptionCookies].asString();
+    if (sessionCookie.buf)
+        [_logic addValue: sessionCookie.asNSString()  forHTTPHeaderField: @"Cookie"];
+    
+    CBL_LOCK(_db) {
```

Comment:
Is this confirmed that we need to lock? If so, can you add an internal method to set cookie in the CBLDatabase class so that we don't try to lock the db outside the CBLDatabase?

Reply:
Yes. confirmed by Jens. Will move the logic